### ISSUE TITLE
STweep fails to format a large Enum

### ISSUE THREAD

**Issue by MotronicAG** [STweep/STweep#86]
**Describe the bug**
STweep fails to format the large enum E_EnginneringUnits with over 1500 members.

**Expected behavior**
Please import the E_EngineeringUnits.XML in TwinCat XAE Shell via PlcOpenXml Import. You will see 3 Enums: E_EngineeringUnits, E_EngineeringUnits_FirstHalf and E_EngineeringUnits_SecondHalf. So, E_EngineeringUnits is devided up in the other two halfs.
 
 - Format E_EngineeringUnits_FirstHalf -> Works
 - Format E_EngineeringUnits_SecondHalf -> Works
 - Format E_EngineeringUnits_FirstHalf -> Works not, CoDeSys Error "STweep - System.OperationCanceledException: The Operation was canceled. at System.Threading.CancellationToken.ThrowOperationCanceledException..."

**STweep version info**
"STweep for TwinCAT XAE Shell 1.0.1" or "STweep for Codesys 1.0.1"

**IDE version info**
TwinCat XAE Shell 4024.55

**Additional context**
Pic with CoDeSys-Error:
![2024-02-26 13_54_53-TcXaeShell](https://github.com/STweep/STweep/assets/122988684/98b2a5af-db18-4219-a983-41fec837b343)

Thanks for the problem analysis!
Ralf

**Comment by Roald87** [comment]
Probably the same reason as #45

**Comment by MotronicAG** [comment]
Yes, that could be. So, if STweep had 2min time, it might work. @Barteling : Would it be possible to implement a progress bar, perhaps with the option to cancel the job? So the user would know how long it will take and whether he wants to cancel the job...

**Comment by Barteling** [comment]
Hi ralf, where can i download the E_EngineeringUnits.XML ? 
Could you also share your settings? this probably make a huge difference

**Comment by MotronicAG** [comment]
Hi Gerhard

Sorry, i forgot the Enums... 🤦‍♀️
Here are the files:
[EnumsAndOptions.zip](https://github.com/STweep/STweep/files/14420188/EnumsAndOptions.zip)
FYI: The duration for the code-formattings on our machine are:
E_EngineeringUnits_1: 18s -> OK
E_EngineeringUnits_2: 58s -> OK
E_EngineeringUnits: Error Message after 60s

Thanks

**Comment by Barteling** [comment]
I believe i fixed this, changes will be reflected in the next release



> Probably the same reason as #45

Unfortunately not, #45 is an edge case i havent found the right solution for.

> Yes, that could be. So, if STweep had 2min time, it might work. @Barteling : Would it be possible to implement a progress bar, perhaps with the option to cancel the job? So the user would know how long it will take and whether he wants to cancel the job...


Yes, i'll consider this

**Comment by Barteling** [comment]
The latest beta has some performance upgrades, can you check if this works in your case?

**Comment by MotronicAG** [comment]
Thank you for your efforts.

But where can i download the beta?

**Comment by Barteling** [comment]
https://www.stweep.com/beta-releases

**Comment by MotronicAG** [comment]
Aaaa. Thanks.

I have tested it again with the E_EngineeringUnits:
Version 4.2.3: Execution aborted without error message after 60s, no formatting performed on the file.
Version 4.3.0 Beta: Execution aborted without error message after 60s, no formatting performed on the file.

So unfortunately no improvement, exactly the same behavior.

Since I have changed the StweepSettings since the last time, here is the current version:
[STweepSettings.zip](https://github.com/STweep/STweep/files/15137180/STweepSettings.zip)

**Comment by MotronicAG** [comment]
Can you please reopen this Bug, because its not fixed.
Thanks!

**Comment by MotronicAG** [comment]
Still the same in Version 4.4.4:
Execution aborted without error message after 60s, no formatting performed on the file.

**Comment by Barteling** [comment]
Improved in 4.5.0-Beta
https://www.stweep.com/beta-releases/
